ui/src/app/applications/components/resource-details/resource-details.tsx

```diff
@@ -162,21 +168,17 @@ export const ResourceDetails = (props: ResourceDetailsProps) => {
                 content: <ApplicationSummary app={application} updateApp={(app, query: {validate?: boolean}) => updateApp(app, query)} />
             },
             {
-                title: 'SOURCES',
-                key: 'sources',
+                title: application.spec.sources === undefined ? 'PARAMETERS' : 'SOURCES',
+                key: 'parameters',
                 content: (
-                    <DataLoader key='appDetails' input={application} load={app => getSources(app)}>
```

Comment:
Not very much familiar with dataloader but don't we need DataLoader? how you are passing the `app`

Reply:
Hi @ashutosh16 , thanks for reviewing this.  In my first stage change (for showing all the sources under that tab in read-only mode, with the Edit button disabled), it is used to load all the sources all at once.  See the call to getSources function, which I've now removed below.  It calls the appDetails api to get all the details of the sources.  If I left it this way, that means all the data is load all at once for all the collapsible sections.

With this new approach, I delayed the loading for each source until the section is expanded.  This improves performance and avoids flickering when any change in the sources occur.   The DataLoader code is moved to application-parameters.tsx. For multiple sources, see line 250.  For the single source, see line 192.  This explains my second bullet point here:  https://github.com/argoproj/argo-cd/pull/17890#issue-2251208515
The Loading message appears within the source card, and not outside all the cards.

Note that when the single source field is 'deprecated' or 'removed', I refactored it so that we could more easily remove the code related to how the single source is handled (even if there is a bit of duplication).  Also, I tried to minimize any potential regressions against the original single source behavior.